django/db/models/query.py

```diff
@@ -1208,17 +1234,63 @@ def _raw_delete(self, using):
 
     _raw_delete.alters_data = True
 
-    def update(self, **kwargs):
+    # Existing signature with keyword only updates.
+    @overload
+    def update(self, **updates) -> int: ...
+
+    # New signature with positional only updates to preserve backward
+    # compatibility with update(updates={"something"}) for models with
+    # an "updates" field.
+    @overload
+    def update(self, updates: dict, /) -> int: ...
+
+    # New signature with positional only updates and returning support which
+    # also allows paves the way for other options in the future if needs be.
+    @overload
+    def update(
+        self, updates: dict, *, returning: list[str] | tuple[str]
+    ) -> list[tuple]: ...
+
+    def update(self, *args, **kwargs):
         """
         Update all elements in the current QuerySet, setting all the given
         fields to the appropriate values.
         """
         self._not_support_combined_queries("update")
         if self.query.is_sliced:
             raise TypeError("Cannot update a query once a slice has been taken.")
+        returning_fields = None
+        if args:
+            update_values = args[0]
+            returning = kwargs.pop("returning", None)
```

Comment:
Explicitly not named `returning_fields` as some backends (e.g. Postgres) allows any expressions to be returned so it could eventually be adapted not only to accept `returning: Iterable[str]` but also `returning: Iterable[str | Resolvable]`.